Add most of the missing pandas methods as stubs.

ueshin opened the pull request:
Added missing pandas methods as stubs and modified the current methods to follow pandas API.

Closes #77.

ueshin commented:
cc @thunterdb

thunterdb commented on databricks/koala/frame.py:
```diff
@@ -65,8 +66,18 @@ def iteritems(self):
     @derived_from(pd.DataFrame)
-    def to_html(self, index=True, classes=None):
-        return self.toPandas().to_html(index=index, classes=classes)
+    def to_html(self, buf=None, columns=None, col_space=None, header=True, index=True,
```
nice, thank you.

thunterdb commented on dev/_make_missing_functions.py:
```diff
@@ -0,0 +1,154 @@
+from databricks.koala.series import PandasLikeSeries
+
+
+def inspect_missing_functions(original_type, target_type):
```
please put some documentation about what these functions are expected to do

thunterdb replied:
and return

thunterdb commented on dev/_make_missing_functions.py:
```diff
@@ -0,0 +1,154 @@
+    return missings, updatings
+
+
+def format_arguments(name, signature):
```
can you explain what this function does?

ueshin commented:
@thunterdb I reverted the changes for the existing methods.
One thing, as for `Series.rename`, the signature is completely different from pandas'.
I guess I referred dask's signature when I introduced the method.
I think we should fix it now. WDYT?

pandas:
```py
def rename(self, index=None, **kwargs):
```

koala:
```py
def rename(self, name, inplace=False):
```

thunterdb commented on databricks/koala/dask/utils.py:
```diff
@@ -62,8 +62,8 @@ def derived_from(original_klass, version=None, ua_args=[]):
     ua_args : list
-        List of keywords which Dask doesn't support. Keywords existing in
-        original but not in Dask will automatically be added.
+        List of keywords which Koala doesn't support. Keywords existing in
```
I forgot. Let's keep the dask code as close as possible to the original code. This will help when trying to backport upstream fixes.

ueshin commented on databricks/koala/dask/utils.py:
```diff
@@ -88,7 +88,7 @@ def wrapper(method):
             if len(not_supported) > 0:
                 note = ("\n        Notes\n        -----\n"
-                        "        Koala doesn't support the following argument(s).\n\n")
+                        "        Dask doesn't support the following argument(s).\n\n")
```
@thunterdb At least this should be `Koala`? otherwise "Dask" appears in the document.

thunterdb commented:
Good point about this message, There is no need to translate all the other
comments.

On Thu, Apr 18, 2019 at 11:22 AM Takuya UESHIN <notifications@github.com>
wrote:

> *@ueshin* commented on this pull request.
> ------------------------------
>
> In databricks/koala/dask/utils.py
> <https://github.com/databricks/spark-pandas/pull/80#discussion_r276583064>
> :
>
> > @@ -88,7 +88,7 @@ def wrapper(method):
>
>              if len(not_supported) > 0:
>                  note = ("\n        Notes\n        -----\n"
> -                        "        Koala doesn't support the following argument(s).\n\n")
> +                        "        Dask doesn't support the following argument(s).\n\n")
>
> @thunterdb <https://github.com/thunterdb> At least this should be Koala?
> otherwise "Dask" appears in the document.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/databricks/spark-pandas/pull/80#pullrequestreview-228182731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZ6GAKL3TLFOATMA2B5FJ3PRA4TXANCNFSM4HGTG3BA>
> .
>
